model/validation.go

```diff
@@ -204,16 +204,353 @@ func (t Tags) Validate() error {
 	return errs
 }
 
+func validateFQDN(fqdn string) error {
```

Comment:
I used multierror to make sure the user sees all fields that are invalid.  This particular method validates a DNS name, and it felt like too much data to receive multiple complaints about the same DNS name.